ggml/src/ggml-vulkan/ggml-vulkan.cpp

```diff
@@ -7433,9 +7433,13 @@ static void ggml_vk_cleanup(ggml_backend_vk_context * ctx) {
 }
 
 static int ggml_vk_get_device_count() {
-    ggml_vk_instance_init();
-
-    return vk_instance.device_indices.size();
+    try {
+        ggml_vk_instance_init();
```

Comment:
Seems like a failure here will leave things in a weird partially-initialized state.

Reply:
We could special-case just the one exception `vk::IncompatibleDriverError` that happens here--under the assumption that that one won't leave it in a partially-initialized state. What do you think?

Reply:
There are two expected cases, failure to initialize the Vulkan instance (issue with the loader) or no devices found (but an instance was created). It would probably be good to handle those inside of `ggml_vk_instance_init()`, to be able to clean up the instance if one was created.